modules/space/rendering/renderablesatellites.cpp

```diff
@@ -74,53 +74,42 @@ namespace {
         "Trail Fade",
         "This value determines how fast the trail fades and is an appearance property. "
     };
+
+    struct [[codegen::Dictionary(RenderableSatellites)]] Parameters {
+        // [[codegen::verbatim(SegmentsInfo.description)]]
+        double segments;
+
+        // [[codegen::verbatim(PathInfo.description)]]
+        std::string path;
+
+        // [[codegen::verbatim(LineWidthInfo.description)]]
+        std::optional<double> lineWidth;
+
+        // [[codegen::verbatim(LineColorInfo.description)]]
+        glm::dvec3 color;
+
+        // [[codegen::verbatim(TrailFadeInfo.description)]]
+        std::optional<double> trailFade;
+    };
+#include "renderablesatellites_codegen.cpp"
 }
 
 namespace openspace {
 
 documentation::Documentation RenderableSatellites::Documentation() {
-    using namespace documentation;
-    return {
-        "RenderableSatellites",
-        "space_renderable_satellites",
-        {
-            {
-                SegmentsInfo.identifier,
-                new DoubleVerifier,
-                Optional::No,
-                SegmentsInfo.description
-            },
-            {
-                PathInfo.identifier,
-                new StringVerifier,
-                Optional::No,
-                PathInfo.description
-            },
-            {
-                LineWidthInfo.identifier,
-                new DoubleVerifier,
-                Optional::Yes,
-                LineWidthInfo.description
-            },
-            {
-                LineColorInfo.identifier,
-                new DoubleVector3Verifier,
-                Optional::No,
-                LineColorInfo.description
-            },
-            {
-                TrailFadeInfo.identifier,
-                new DoubleVerifier,
-                Optional::Yes,
-                TrailFadeInfo.description
-            }
-        }
-    };
+    documentation::Documentation doc = codegen::doc<Parameters>();
+    doc.id = "space_renderable_satellites";
+    return doc;
 }
 
 RenderableSatellites::RenderableSatellites(const ghoul::Dictionary& dictionary)
     : RenderableOrbitalKepler(dictionary)
-{}
+{
+    // Commented out right now as its not super clear how it works with inheritance. We'd
+    // probably want a codegen::check function that only does the checking without
+    // actually creating a Parameter objects
```

Comment:
Do you mean that the inheritance of `RenderableSatellites : public RenderableOrbitalKepler` is not super clear, or are you referring to how codegen works with inheritance?

Reply:
It's not really clear to me how we want to use the codegen in the case of inheritances... Since we can't (i think) expose the tagged `Parameters` struct to be used in a different type , we would have to duplicate a lot of the parameters, which feels icky

Reply:
Agreed, that sounds (tr)icky